tests/MongoDB.Driver.Tests/Specifications/transactions/TransactionTestRunner.cs

```diff
@@ -117,6 +137,37 @@ private void Run(BsonDocument shared, BsonDocument test)
             }
         }
 
+        /// <summary>
+        /// Temporary patch until SERVER-39704 is resolved.
+        /// </summary>
+        private void PrimeShardRoutersWithDistinctCommand()
+        {
+            var connectionString = CoreTestConfiguration.ConnectionStringWithMultipleShardRouters.ToString();
+            var clientSettings = MongoClientSettings.FromUrl(new MongoUrl(connectionString));
+            var shardRouters = clientSettings.Servers.ToList();
+            MongoClientSettings CloneSettingsWithOneServer(MongoClientSettings settings, MongoServerAddress server)
+            {
+                var clone = settings.Clone();
+                clone.Server = server;
+                return clone;
+            }
+
+            var firstShardRouterClient = new MongoClient(CloneSettingsWithOneServer(clientSettings, shardRouters[0]));
```

Comment:
This seems brittle.  At least assert that shardRouters length == 2 so future failures are obvious to debug (i.e. when we change the number of shard routers in mongo orchestration)

Reply:
Or just make this a loop

Reply:
Agreed on the brittleness. I've taken your suggestions Jeff while also refactoring per my offline discussion with Robert!